crest/Assets/Crest/Crest/Scripts/OceanRenderer.cs

```diff
@@ -319,6 +358,15 @@ void OnEnable()
             Instance = this;
             Scale = Mathf.Clamp(Scale, _minScale, _maxScale);
 
+            _bufPerCascadeInstanceData = new ComputeBuffer(_perCascadeInstanceData.Length, Marshal.SizeOf<PerCascadeInstanceData>());
+            Shader.SetGlobalBuffer("_PerCascadeInstanceData", _bufPerCascadeInstanceData);
+
+            _bufCascadeDataTgt = new ComputeBuffer(_cascadeParamsTgt.Length, Marshal.SizeOf<CascadeParams>());
```

Comment:
UnsafeUtilty.SizeOf is the preferred way of getting the size of structs in Unity. You should avoid using Marshal.

https://docs.unity3d.com/ScriptReference/Unity.Collections.LowLevel.Unsafe.UnsafeUtility.SizeOf.html

The Marshal functions are slower, generate garbage, and the UnsafeUtility functions optimise better when compiled to IL2CPP.

Reply:
Ok thx will change it later

Reply:
I could not seem to use it, the namespace seemed to be there but the utility UnsafeUtility did not seem to resolve for me. Would it be possible to get an assist with this?

Reply:
Done :)